pg_lakehouse/README.md

```diff
@@ -0,0 +1,299 @@
+
+```sql
+CREATE FOREIGN TABLE hits (
+  "EventDate" INTEGER
+) USING foreign_server OPTIONS (...);
+
+SELECT to_date("EventDate") FROM hits LIMIT 1;
+```
+
+### Timestamp
+
+Similarly, the `to_timestamp` function converts an integer representing the number of seconds elapsed since the UNIX epoch to a Postgres `TIMESTAMP` type.
+
+```sql
+CREATE FOREIGN TABLE hits (
+  "EventTime" BIGINT
+) USING foreign_server OPTIONS (...);
+
+SELECT to_timestamp("EventTime") FROM hits LIMIT 1;
+```
+
+## Development
+
+### Install Rust
+
+To develop the extension, first install Rust via `rustup`.
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+rustup install <version>
+
+rustup default <version>
+```
+
+Note: While it is possible to install Rust via your package manager, we recommend using `rustup` as we've observed inconcistencies with Homebrew's Rust installation on macOS.
+
+Then, install the PostgreSQL version of your choice using your system package manager. Here we provide the commands for the default PostgreSQL version used by this project:
+
+### Install Postgres
+
+```bash
+# macOS
+brew install postgresql@16
+
+# Ubuntu
+wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
+sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'
+sudo apt-get update && sudo apt-get install -y postgresql-16 postgresql-server-dev-16
+```
+
+If you are using Postgres.app to manage your macOS PostgreSQL, you'll need to add the `pg_config` binary to your path before continuing:
+
+```bash
+export PATH="$PATH:/Applications/Postgres.app/Contents/Versions/latest/bin"
+```
+
+### Install pgrx
+
+Then, install and initialize `pgrx`:
+
+```bash
+# Note: Replace --pg16 with your version of Postgres, if different (i.e. --pg15, --pg14, etc.)
+cargo install --locked cargo-pgrx --version 0.11.3
+
+# macOS arm64
+cargo pgrx init --pg16=/opt/homebrew/opt/postgresql@16/bin/pg_config
+
+# macOS amd64
+cargo pgrx init --pg16=/usr/local/opt/postgresql@16/bin/pg_config
+
+# Ubuntu
+cargo pgrx init --pg16=/usr/lib/postgresql/16/bin/pg_config
+```
+
+If you prefer to use a different version of Postgres, update the `--pg` flag accordingly.
+
+Note: While it is possible to develop using pgrx's own Postgres installation(s), via `cargo pgrx init` without specifying a `pg_config` path, we recommend using your system package manager's Postgres as we've observed inconsistent behaviours when using pgrx's.
+
+### Adding a Service
+
+`pg_lakehouse` uses Apache OpenDAL to integrate with various object stores. As of the time of writing, some — but not all — of the object stores supported by OpenDAL have been integrated.
+
+Adding support for a new object store is as straightforward as creating a file in the `fdw/` folder that implements the `BaseFdw` trait.
```

Comment:
This README is solid but it's also very long. We might want to move some of the examples over to our Docs at some point soon